src/Security/ForbiddenDownloadStrategy.php

```diff
@@ -15,20 +15,37 @@
 
 use Sonata\MediaBundle\Model\MediaInterface;
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Translation\TranslatorInterface as LegacyTranslatorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
 
 /**
  * @final since sonata-project/media-bundle 3.21.0
  */
 class ForbiddenDownloadStrategy implements DownloadStrategyInterface
 {
     /**
-     * @var TranslatorInterface
+     * @var LegacyTranslatorInterface|TranslatorInterface
      */
     protected $translator;
 
-    public function __construct(TranslatorInterface $translator)
+    public function __construct(object $translator)
     {
+        if (!$translator instanceof TranslatorInterface) {
+            @trigger_error(sprintf(
+                'The $translator parameter should be an instance of "%s" and will be required in 4.0.',
+                TranslatorInterface::class
+            ), \E_USER_DEPRECATED);
+        }
+
+        if (!$translator instanceof TranslatorInterface && !$translator instanceof LegacyTranslatorInterface) {
+            throw new \TypeError(sprintf(
+                'Argument 1 must be an instance of "%s" or preferably "%s", "%s" given.',
```

Comment:
IMO, the preferred argument is indicated by the previous deprecation message.
```suggestion
                'Argument 1 must be an instance of "%s" or "%s", "%s" given.',
```